lib/bundler/audit/cli/formats/junit.rb

```diff
@@ -0,0 +1,118 @@
+#
+# Copyright (c) 2013-2021 Hal Brodigan (postmodern.mod3 at gmail.com)
+#
+# bundler-audit is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# bundler-audit is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with bundler-audit.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+require 'thor'
+
+module Bundler
+  module Audit
+    class CLI < ::Thor
+      module Formats
+        module Junit
+          #
+          # Prints any findings as an XML junit report.
+          #
+          # @param [Report] report
+          #   The results from the {Scanner}.
+          #
+          # @param [IO, File] output
+          #   Optional output stream.
+          #
+          def print_report(report, output=$stdout)
+            original_stdout = $stdout
+            $stdout = output
+
+            print_xml_testsuite(report) do
+              report.each do |result|
+                print_xml_testcase(result)
+              end
+            end
+
+            $stdout = original_stdout
+          end
+
+          private
+
+          def print_xml_testsuite(report)
+            say <<~XML
+              <?xml version="1.0" encoding="UTF-8" ?>
+              <testsuites id="#{ Time.now.to_i }" name="Bundle Audit">
+                <testsuite id="Gemfile" name="Ruby Gemfile" failures="#{ report.count }">
+            XML
+
+            yield
+
+            say <<~XML
+                </testsuite>
+              </testsuites>
+            XML
+          end
+
+          def print_xml_testcase(result)
+            case result
+            when Results::InsecureSource
+              say <<~XML
+                  <testcase id="#{ result.source }" name="Insecure Source URI found: #{result.source}">
+                    <failure message="Insecure Source URI found: #{result.source}" type="Unknown"></failure>
+                  </testcase>
+              XML
+            when Results::UnpatchedGem
+              say <<~XML
+                  <testcase id="#{ result.gem.name }" name="#{ bundle_title(result) }">
+                    <failure message="#{ result.advisory.title }" type="#{ result.advisory.criticality }">
```

Comment:
Text fields should probably be wrapped in `CGI.escapeHTML()` to prevent any quote escaping or other invalid XML characters getting into the XML.

Reply:
Probably could define a `h` or `xml` method and escape any data that goes into the XML, just in case.